File: vercel/api/webhook.js
const feishu = require('./feishu');
const bilibili = require('./bilibili');
const minimax = require('./minimax');

const processedEvents = new Set();

module.exports = async (req, res) => {
  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }
  
  const data = req.body;
  
  if (data.challenge) {
    return res.json({ challenge: data.challenge });
  }
  
  if (data.header && data.header.event_type === 'im.message.receive_v1') {
    const eventId = data.header.event_id;
    
    if (processedEvents.has(eventId)) {
      return res.json({ code: 0 });
    }
    processedEvents.add(eventId);
    
    if (processedEvents.size > 1000) {
      processedEvents.clear();
    }
    
    const event = data.event;
    const message = event.message || {};
    const chatId = message.chat_id || '';
    const content = JSON.parse(message.content || '{}');
    const text = content.text || '';
    
    const bvidResult = bilibili.extractBvid(text);
    let bvid = null;
    
    if (bvidResult && bvidResult.isShortUrl) {
      const fullUrl = await bilibili.resolveShortUrl(bvidResult.url);
      const match = fullUrl.match(/BV[a-zA-Z0-9]+/);
      if (match) bvid = match[0];
    } else if (bvidResult) {
      bvid = bvidResult;
    }
    
    if (bvid) {
      const token = await feishu.getToken();
      const result = await bilibili.getSubtitle(bvid);
      
      if (result.error) {
        await feishu.sendMessage(token, chatId, `提取失败: ${result.error}`);
      } else {
        const summary = await minimax.summarize(result.subtitle, '请总结这段字幕的核心观点，用简洁的中文回答，不超过100字');
        
        const now = new Date();
        const timestamp = now.getTime();
        
        await feishu.writeToBase(token, {
          '视频标题': result.title,
          'BV号': bvid,
          '字幕内容': result.subtitle.substring(0, 2000),
          '抓取时间': timestamp,
          '创建时间': timestamp,
          '作者': result.author,
          '封面图': result.cover,
          '简介': result.desc,
          'minimax总结': summary
        });
        
        await feishu.sendMessage(token, chatId, `字幕提取完成\n视频: ${result.title}\n作者: ${result.author}\n\nAI总结:\n${summary}`);
      }
    }
  }
  
  res.json({ code: 0 });
};

File: app/routes/webhook.py
from flask import Blueprint, request, jsonify
from ..services.feishu import FeishuService
from ..services.bilibili import BilibiliService
from ..services.minimax import MinimaxService

webhook_bp = Blueprint("webhook", __name__)

feishu = FeishuService()
bilibili = BilibiliService()
minimax = MinimaxService()

processed_events = set()


@webhook_bp.route("/webhook", methods=["POST"])
def webhook():
    data = request.json
    
    if "challenge" in data:
        return jsonify({"challenge": data["challenge"]})
    
    if "header" in data and data["header"].get("event_type") == "im.message.receive_v1":
        event_id = data["header"].get("event_id", "")
        
        if event_id in processed_events:
            return jsonify({"code": 0})
        processed_events.add(event_id)
        
        if len(processed_events) > 1000:
            processed_events.clear()
        
        event = data["event"]
        message = event.get("message", {})
        chat_id = message.get("chat_id", "")
        content = __import__("json").loads(message.get("content", "{}"))
        text = content.get("text", "")
        
        bvid = bilibili.extract_bvid(text)
        if bvid:
            token = feishu.get_token()
            result, error = bilibili.get_subtitle(bvid)
            
            if error:
                feishu.send_message(token, chat_id, f"提取失败: {error}")
            else:
                summary = minimax.summarize(result["subtitle"], "请总结这段字幕的核心观点，用简洁的中文回答，不超过100字")
                
                from datetime import datetime
                now = datetime.now()
                timestamp_ms = int(now.timestamp() * 1000)
                
                feishu.write_to_base(token, {
                    "视频标题": result["title"],
                    "BV号": bvid,
                    "字幕内容": result["subtitle"][:2000],
                    "抓取时间": timestamp_ms,
                    "创建时间": timestamp_ms,
                    "作者": result["author"],
                    "封面图": result["封面图"],
                    "简介": result["desc"],
                    "minimax总结": summary
                })
                
                feishu.send_message(token, chat_id, f"字幕提取完成\n视频: {result['title']}\n作者: {result['author']}\n\nAI总结:\n{summary}")
    
    return jsonify({"code": 0})
